### PR TITLE
feat: diagnostics tool implementation

### PR DESCRIPTION
**Pull request by krmcbride**
I've been tinkering around with fully implementing the `getDiagnostics` tool and have things basically working in a similar fashion to how they do in VSCode.  

One little improvement I've made over the VSCode behavior is, when you ask about diagnostics in a file that doesn't have a buffer open, rather than lying to you and telling you the file is clean, it'll return a MCP error to indicate it can't return diagnostics for files without a buffer.

 ##  Summary

  Add getDiagnostics tool for retrieving LSP diagnostics from Neovim

  ## Details

  - Implements MCP-compatible getDiagnostics tool that retrieves LSP diagnostics (errors, warnings) from open
  buffers
  - Supports optional uri parameter to filter diagnostics for a specific file
  - Returns diagnostics in MCP content format with JSON-encoded diagnostic information
  - Converts 0-indexed Neovim diagnostic positions to 1-indexed for better user experience (matching editor line
  numbers)
  - Includes comprehensive error handling for unopened files and missing LSP/diagnostic APIs
  - Adds full test coverage for all functionality including URI filtering and edge cases

### CODE REVIEW

File: lua/claudecode/tools/get_diagnostics.lua [hunk 1]
```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
...
+    -- Get buffer number for the specific file
+    local bufnr = vim.fn.bufnr(filepath)
+    if bufnr == -1 then
+      -- File is not open in any buffer, throw an error
+      log.debug("File buffer must be open to get diagnostics: " .. filepath)
+      error({
+        code = -32001,
+        message = "File not open in buffer",
+        data = "File must be open in Neovim to retrieve diagnostics: " .. filepath,
+      })
```
Comment (ThomasK33) [inline lua/claudecode/tools/get_diagnostics.lua:70]: Note to myself: Refactor the error returning in this file, as in the rest of the codebase, we return errors as values.

File: lua/claudecode/tools/get_diagnostics.lua [hunk 2]
```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
...
+    -- Get diagnostics for all buffers
+    log.debug("Getting diagnostics for all open buffers")
+    diagnostics = vim.diagnostic.get()
+  else
+    -- Remove file:// prefix if present
+    local uri = params.uri
+    local filepath = uri
+    if uri:sub(1, 7) == "file://" then
+      filepath = uri:sub(8) -- Remove "file://" prefix
+    end
```
Comment (ThomasK33) [inline lua/claudecode/tools/get_diagnostics.lua:59]: Can we use `vim.uri_to_fname` here instead?

File: lua/claudecode/tools/get_diagnostics.lua [hunk 3]
```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
...
+
+  log.debug("getDiagnostics handler called with params: " .. vim.inspect(params))
+
+  -- Extract the uri parameter
+  local diagnostics
+
+  if not params.uri then
+    -- Get diagnostics for all buffers
+    log.debug("Getting diagnostics for all open buffers")
+    diagnostics = vim.diagnostic.get()
```
Comment (ThomasK33) [inline lua/claudecode/tools/get_diagnostics.lua:52]: `vim.diagnostic.get(nil)` to make “all buffers” explicit.

File: lua/claudecode/tools/get_diagnostics.lua [hunk 4]
```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
...
+        text = vim.fn.json_encode({
+          -- Use the file path and diagnostic information
+          filePath = file_path,
+          -- Convert line and column to 1-indexed
+          line = diagnostic.lnum + 1,
+          character = diagnostic.col + 1,
+          severity = diagnostic.severity, -- e.g., vim.diagnostic.severity.ERROR
+          message = diagnostic.message,
+          source = diagnostic.source,
+        }),
```
Comment (ThomasK33) [inline lua/claudecode/tools/get_diagnostics.lua:95]: Prefer `vim.json.encode` over `vim.fn.json_encode` for speed and no Vim-Lua round-trip.

File: lua/claudecode/tools/get_diagnostics.lua [hunk 5]
```diff
@@ -17,29 +32,78 @@ local function handler(_params) -- Prefix unused params with underscore
...
 
-  local all_diagnostics = vim.diagnostic.get(0) -- Get for all buffers
+  local log_module_ok, log = pcall(require, "claudecode.logger")
+  if not log_module_ok then
+    return {
+      code = -32603, -- Internal error
+      message = "Internal error",
+      data = "Failed to load logger module",
+    }
+  end
```
Comment (ThomasK33) [inline lua/claudecode/tools/get_diagnostics.lua:42]: I don't think that we have to `pcall` this.

**Review by ThomasK33 — COMMENTED** [review]
Thanks for opening this PR; it looks very nice.

Please make sure to fix the luacheck errors so that CI can pass.

Also left some review comments on the changes.

File: lua/claudecode/tools/get_diagnostics.lua (hunk 2 above)
Comment (krmcbride) [reply lua/claudecode/tools/get_diagnostics.lua]: Done.

Since `uri_to_fname` throws if the input doesn't have a scheme I went back and forth on how defensive to be, but ended up going with the simplest/least defensive approach of just blindly passing whatever `uri` is into it assuming it will always be a proper file URI.

**Comment by krmcbride** [comment]
Something interesting happened as I was testing this just now -- Claude started adding 1 to line and character numbers on its own, so everything it mentioned was off by one (since I'm already compensating for this).  So I asked it what the hell it was doing (because I could see the MCP diag tool response body was accurate) and it said that since it was Neovim LSP diagnostics, and those are 0-indexed, it was adding 1 to everything for me...

So... I've removed any mention of neovim LSP from the things Claude can see in the diagnostic tool 😆 

We're living in a weird new world

File: lua/claudecode/tools/get_diagnostics.lua (hunk 2 above)
Comment (ThomasK33) [reply lua/claudecode/tools/get_diagnostics.lua]: That's fine, simple is good for now.
If we're trying to be defensive here, we might also wrap it in a pcall and check if it's ok.

File: lua/claudecode/tools/get_diagnostics.lua (hunk 5 above)
Comment (ThomasK33) [reply lua/claudecode/tools/get_diagnostics.lua]: In that case, I'll have to take a look and do some refactoring. Thanks for the mention 👍

**Comment by ThomasK33** [comment]
You'll need to update the tests, as they are still expecting the old string values containing Neovim references.
Apart from that, it looks good to me.

**Comment by krmcbride** [comment]
> You'll need to update the tests, as they are still expecting the old string values containing Neovim references. Apart from that, it looks good to me.

Oops! OK, should be g2g

**Review by ThomasK33 — APPROVED** [review]
